### PR TITLE
Fix: SQL schema creation with MySQL.

### PR DESCRIPTION
**Pull request by julmud**
Also fix some ugly undefined variable warning when the
connection to the database can't be established.

Fixes issue #7

### CODE REVIEW

File: global.php [hunk 1]
```diff
@@ -706,6 +706,7 @@ function StripMagicQuotes(&$var) {
 $db = new sql_db($dbhost, $dbuser, $dbpasswd, $dbname, $dbport, false, $debugSQL, true);
 $db_Errors = $db->GetErrorState();
 
+$collectiontypelist = array();
```
Comment (gjnoonan) [inline global.php:709]: Ah good, I was about to push this, I'd fixed it but forgot to move the patch into #4.

This change should be moved into a separate commit though, the whitespace removal is fine to be in either commit but this change warrants it's own commit, and explanation as to why.

**Review by gjnoonan — CHANGES_REQUESTED** [review]
I have Tested this on MySQL 5.6, 5.7, and no regressions were noticed. I could import/update fine as well as view the collection.

I also tried this on MySQL 8.0, it still does not work but requires changes outside of the scope of this patch.

Just need the extra commit, and then I'll approve :-)

File: global.php (hunk 1 above)
Comment (julmud) [reply global.php:709]: Done. :-)

**Comment by julmud** [comment]
> I also tried this on MySQL 8.0, it still does not work but requires changes outside of the scope of this patch.

What problems did you encounter with MySQL 8.0? I just ran the script with MySQL 8.0.18 (through phpMyAdmin) on an empty schema and only had (normal) notices for the `DROP TABLE IF EXISTS` queries.

**Comment by gjnoonan** [comment]
> What problems did you encounter with MySQL 8.0? I just ran the script with MySQL 8.0.18 (through phpMyAdmin) on an empty schema and only had (normal) notices for the `DROP TABLE IF EXISTS` queries.

The errors I had were when running an update, it threw SQL errors when trying to insert into `DVDPROFILER_dvd` and `DVDPROFILER_common_actor`

```
Incorrect string value: '\xA324.99' for column 'srp' at row 1
```

```
INSERT INTO DVDPROFILER_dvd (id,upc,builtinmediatype,custommediatype,mediabannerfront,mediabannerback,title,sorttitle,originaltitle,description,countryoforigin,countryoforigin2,countryoforigin3,region,collectiontype,realcollectiontype,auxcolltype,collectionnumber,rating,ratingsystem,ratingage,ratingvariant,ratingdetails,productionyear,released,runningtime,casetype,caseslipcover,isadulttitle,formatvideostandard,formatletterbox,format16x9,formataspectratio,formatcolorcolor,formatcolorbw,formatcolorcolorized,formatcolormixed,formatpanandscan,formatfullframe,formatdualsided,formatduallayered,dim2d,dim3danaglyph,dim3dbluray,featuresceneaccess,featuretrailer,featurebonustrailers,featuremakingof,featurecommentary,featuredeletedscenes,featureinterviews,featureouttakes,featurestoryboardcomparisons,featurephotogallery,featureproduction in /var/www/mysqli.php on line 456
```

```

INSERT INTO DVDPROFILER_dvd (id,upc,builtinmediatype,custommediatype,mediabannerfront,mediabannerback,title,sorttitle,originaltitle,description,countryoforigin,countryoforigin2,countryoforigin3,region,collectiontype,realcollectiontype,auxcolltype,collectionnumber,rating,ratingsystem,ratingage,ratingvariant,ratingdetails,productionyear,released,runningtime,casetype,caseslipcover,isadulttitle,formatvideostandard,formatletterbox,format16x9,formataspectratio,formatcolorcolor,formatcolorbw,formatcolorcolorized,formatcolormixed,formatpanandscan,formatfullframe,formatdualsided,formatduallayered,dim2d,dim3danaglyph,dim3dbluray,featuresceneaccess,featuretrailer,featurebonustrailers,featuremakingof,featurecommentary,featuredeletedscenes,featureinterviews,featureouttakes,featurestoryboardcomparisons,featurephotogallery,featureproductionnotes,featuredvdromcontent,featuregame,featuremultiangle,feat in /var/www/mysqli.php on line 456
```

The errors are a bit thin. I will look into it, but I don't think it is a blocker for this MR going in, since we don't currently work with MYSQL8 anyway.

**Comment by julmud** [comment]
> The errors are a bit thin. I will look into it, but I don't think it is a blocker for this MR going in, since we don't currently work with MYSQL8 anyway.

I think I found the cause: every time a SQL query uses a "direct" string, we now need to specify its charset. I.e. we have to write:
`select * from dvd where title = _latin1'Valérian et la Cité des Mille Planètes'`
instead of:
`select * from dvd where title = 'Valérian et la Cité des Mille Planètes'`

This will be painful to implement, as SQL queries are generated nearly everywhere in the code...

I'll open a new issue to track this.
